clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -856,16 +896,13 @@ protected void leaderExecuteTopicSwitch(
         ? partitionConsumptionState
             .getLatestProcessedUpstreamRTOffset(OffsetRecord.NON_AA_REPLICATION_UPSTREAM_OFFSET_MAP_KEY)
         : OffsetRecord.LOWEST_OFFSET;
-
-    if (upstreamStartOffset < 0) {
-      if (topicSwitch.rewindStartTimestamp > 0) {
-        upstreamStartOffset = getTopicPartitionOffsetByKafkaURL(
-            newSourceKafkaServer,
-            newSourceTopicPartition,
-            topicSwitch.rewindStartTimestamp);
-      } else {
-        upstreamStartOffset = OffsetRecord.LOWEST_OFFSET;
-      }
+    if (upstreamStartOffset < 0 && newSourceTopic.isRealTime()) {
+      upstreamStartOffset = calculateLeaderUpstreamOffsetWithTopicSwitch(
+          partitionConsumptionState,
```

Comment:
Could you please elaborate? Why do we replace rewind time check with source topic check?

Reply:
Ehh, what do you mean by this? 
This method is the logic to calculate rewind offset based on rewind time fetched from TS .

Reply:
You added this condition: newSourceTopic.isRealTime().
Is it intentional?

Reply:
Yeah, we only need to refer to TS to calculate the rewind offset, if we are switching into a RT leader topic as a leader.